In [1]:
import numpy as np
import pandas as pd
import matplotlib as plt

In [7]:
import plotly.express as px

## Followers

In [38]:
df_followers = pd.read_csv('linkedin_data/followers_job.csv')
df_followers

Job function  Total followers
0                        Engineering              166
1               Business Development              155
2                           Research              117
3                          Education               80
4             Information Technology               47
5                          Marketing               47
6                         Operations               39
7                    Arts and Design               34
8                              Sales               34
9      Community and Social Services               33
10           Media and Communication               33
11    Program and Project Management               32
12                        Consulting               30
13                    Administrative               22
14                Product Management               21
15                   Human Resources               17
16                           Finance               13
17      Customer Success and Support               10
18                  Entrepreneurship                9
19                        Accounting                8
20               Healthcare Services                8
21                             Legal                8
22                        Purchasing                4
23  Military and Protective Services                3
24                 Quality Assurance                3
25                       Real Estate                2

In [39]:
total_followers = df_followers['Total followers'].sum()

# Calculate the percentage of each 'Job function'
df_followers['Percentage'] = (df_followers['Total followers'] / total_followers) * 100

In [45]:
df_followers.head(10)

Job function  Total followers  Percentage
0                    Engineering              166   17.025641
1           Business Development              155   15.897436
2                       Research              117   12.000000
3                      Education               80    8.205128
4         Information Technology               47    4.820513
5                      Marketing               47    4.820513
6                     Operations               39    4.000000
7                Arts and Design               34    3.487179
8                          Sales               34    3.487179
9  Community and Social Services               33    3.384615

The job functions of "Engineering" and "Business Development" have the highest total number of followers, accounting for approximately 17.03% and 15.90% of the total followers, respectively. These two categories appear to be the most popular among the LinkedIn audience in the dataset. The "Research" and "Education" job functions also demonstrate a notable presence on LinkedIn, with around 12.00% and 8.21% of the total followers, respectively. Tech-Related Functions Hold Influence: "Information Technology" and "Marketing" job functions each have around 4.82% of the total followers, highlighting the significance of tech-related and marketing roles in the LinkedIn community.

## Content Metrics

In [29]:
df_posts = pd.read_csv('linkedin_data/posts.csv')
df_posts

Post title  \
0                                                 NaN   
1   Wondering what are latest technology trends? K...   
2                                                 NaN   
3   Business innovations are based on such applied...   
4   💜 We deeply appreciate the dedication of these...   
..                                                ...   
70  Welcome to Benjamin, who is our Lead Product D...   
71  We would like to welcome Rim, our SMM and Mark...   
72                                                NaN   
73  Please take part in our 3-5 minute survey - We...   
74  Hello from RESEARCHPRENEUERS! \n\nToday Alevti...   

                                            Post link Post type  \
0   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
1   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
2   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
3   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
4   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
..                                                ...       ...   
70  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
71  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
72  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
73  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
74  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   

    Campaign name                         Posted by Created date  \
0             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/21/2023   
1             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/20/2023   
2             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/13/2023   
3             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/08/2023   
4             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/02/2023   
..            ...                               ...          ...   
70            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/29/2022   
71            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/28/2022   
72            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/26/2022   
73            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/23/2022   
74            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/21/2022   

    Campaign start date  Campaign end date       Audience  Impressions  \
0                   NaN                NaN  All followers           26   
1                   NaN                NaN  All followers          204   
2                   NaN                NaN  All followers           43   
3                   NaN                NaN  All followers          536   
4                   NaN                NaN  All followers           91   
..                  ...                ...            ...          ...   
70                  NaN                NaN  All followers          109   
71                  NaN                NaN  All followers          146   
72                  NaN                NaN  All followers          118   
73                  NaN                NaN  All followers          160   
74                  NaN                NaN  All followers          130   

    Views (Excluding offsite video views)  Offsite Views  Clicks  \
0                                     NaN            NaN       6   
1                                     NaN            NaN       5   
2                                     NaN            NaN       2   
3                                     NaN            NaN       9   
4                                     NaN            NaN       4   
..                                    ...            ...     ...   
70                                    NaN            NaN       5   
71                                    NaN            NaN       4   
72                                    NaN            NaN       8   
73                                    NaN            NaN       6   
74                                    NaN            NaN       4   

    Click through rate (CTR)  

In [30]:
# drop repost posts
df_posts.dropna(subset=['Post title'], inplace=True)
df_posts.reset_index(drop=True, inplace=True)

In [35]:
df_posts.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 20 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Post title                             62 non-null     object 
 1   Post link                              62 non-null     object 
 2   Post type                              62 non-null     object 
 3   Campaign name                          0 non-null      float64
 4   Posted by                              62 non-null     object 
 5   Created date                           62 non-null     object 
 6   Campaign start date                    0 non-null      float64
 7   Campaign end date                      0 non-null      float64
 8   Audience                               62 non-null     object 
 9   Impressions                            62 non-null     int64  
 10  Views (Excluding offsite video views)  16 non-null     float64
 11  Offsite 

In the dataset, we have a total of 62 posts. Now, let's examine the performance of these posts. On LinkedIn, post performance can be evaluated by analyzing key metrics like impressions, engagement rate, likes, comments, and reposts. Impressions on LinkedIn refer to the number of times a post has been displayed in users' feeds. High impressions indicate that the post is being seen by a larger audience. The engagement rate on LinkedIn is calculated by comparing the total interactions (likes, comments, shares, and clicks) to the total number of impressions. A higher engagement rate signifies that the post is capturing the interest of viewers and generating meaningful interactions.

In [20]:
def plot_histogram(data, x_lable, chart_title):
    fig = px.histogram(data, 
                   x = x_lable,
                   title = chart_title,
                   height = 450)
    fig.show()

In [33]:
# plot histogram of impressions
plot_histogram(df_posts['Impressions'], "Impressions", "Distribution of Impressions")

In [34]:
# plot histogram of impressions
plot_histogram(df_posts['Engagement rate'], "Engagement rate","Distribution of Engagement Rate")

It can be seen from the chart that the majority of posts receive approximately 100 to 300, which suggests that these posts are reaching a moderate audience on LinkedIn impressions. With an engagement rate of less than 0.2, it appears that there is room for improvement in capturing the audience's interest and encouraging more interactions with the posts

In [37]:
df_posts[df_posts['Impressions'] > 400]

Post title  \
1   Business innovations are based on such applied...   
7   🚀 Founders of RESEARCHPRENEURS, Alevtina and J...   
43  That was the #websummit2022 in #Lisbon!\nVery ...   

                                            Post link Post type  \
1   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
7   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
43  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   

    Campaign name                         Posted by Created date  \
1             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/08/2023   
7             NaN  Alevtina Evgrafova, PhD, ACC ICF   05/17/2023   
43            NaN                        Rim Silini   11/07/2022   

    Campaign start date  Campaign end date       Audience  Impressions  \
1                   NaN                NaN  All followers          536   
7                   NaN                NaN  All followers          419   
43                  NaN                NaN  All followers          956   

    Views (Excluding offsite video views)  Offsite Views  Clicks  \
1                                     NaN            NaN       9   
7                                     NaN            NaN      34   
43                                    NaN            NaN     141   

    Click through rate (CTR)  Likes  Comments  Reposts  Follows  \
1                   0.016791      5         0        0      NaN   
7                   0.081146      9         0        2      NaN   
43                  0.147490     20         0        0      NaN   

    Engagement rate Content Type  
1          0.026119          NaN  
7          0.107399          NaN  
43         0.167364          NaN